base_optimization.php

```diff
@@ -0,0 +1,284 @@
+<?php
+
+abstract class base_optimization
+{
+  public $concrete_network_setting;
+
+  /** determines optimization type. has to be set in sub classes*/
+  protected $opt_type;
+
+  /** array mit allen Kampagnen die nicht durch die Filter gefallen sind */
+  protected $data;
+
+  /** current campaign that is being checked/optimized */
+  public $row;
+
+  /**
+   * @var MDB2_Driver_Mysql
+   */
+  public $db;
+
+  /** rpc Objects */
+  protected $campObj, $campOptimization, $campLandingpages, $contCost;
+
+  /** When did the optimization take place */
+  public $optimization_runtime;
+
+  /** Posttracking Settings */
+  protected $period = '0';
+
+  protected $group = 'banner';
+
+  protected $type = 'html';
+
+  protected $debugmode = false;
+
+  /**
+   * Activates or deactivates debug mode
+   *
+   * @param boolean $mode
+   */
+  public function setDebugMode($mode)
+  {
+    $this->debugmode = $mode;
+  }
+
+  public function __construct()
+  {
+    $this->campLandingpages = adition_base::getInstance('campaignlandingpages');
+    $this->campObj = adition_base::getInstance('campaigns');
+    $this->campOptimization = adition_base::getInstance('campaignoptimizations');
+    $this->contCost = adition_base::getInstance('contentunitcosts');
+  }
+
+  public function get_posttracking_report()
```

Comment:
It would be good to change the method name to camelCase and to have a single responsible principle, break down this Method to smaller functions to enhance the readability, maintainability and single responsible principle.
It is good to add some comments to make it more readable.